mitodl/micromasters
# Added task to generate course certificates

#### What are the relevant tickets?
Closes #3460 

#### What's this PR do?
Adds a scheduled task to generate course certificates

#### How should this be manually tested?
Certificates should be generated if a user has a passing `FinalGrade`, and either of the following are true:
- There are no exam runs scheduled for the given course
- There are exam runs for the given course, and the user has a passing `ProctoredExamGrade` for one of them

This script should generate 2 different certificates.
```python
import factory
from courses.models import Program
from exams.factories import ExamRunFactory
from grades import tasks
from grades.factories import (
    FinalGradeFactory,
    ProctoredExamGradeFactory,
)
from grades.models import MicromastersCourseCertificate
from django.contrib.auth.models import User

# Get your user and the financial aid-enabled program
user = User.objects.get(username='YOUR_USERNAME')
fa_program = Program.objects.filter(financial_aid_availability=True).first()
fa_courses = fa_program.course_set.all()
course_without_exam = fa_courses[0]
course_with_exam = fa_courses[1]

# Create passing FinalGrades
FinalGradeFactory.create(user=user, course_run=course_without_exam.courserun_set.first(), passed=True)
FinalGradeFactory.create(user=user, course_run=course_with_exam.courserun_set.first(), passed=True)

# Create an exam run for one of the courses, along with a passing ProctoredExamGrade
exam_run = ExamRunFactory.create(course=course_with_exam)
ProctoredExamGradeFactory.create(
    user=user, 
    course=course_with_exam, 
    exam_run=exam_run,
    passed=True
)

# Run the task
tasks.generate_course_certificates_for_fa_students.delay()

# After the task finished, see the generated certificates
MicromastersCourseCertificate.objects.all()
```

You will also need to have a program page in the CMS and at least one certificate signatory associated with the course (model: `CourseCertificateSignatories`). Navigate to `/cms/`, click "Explorer" and select the home page (this should be the only option that comes up). If you don't have a program page for "Digital Learning", click "Add Child Page" and create the program page for that program. At the bottom of the form to create/edit a program page, you should see "Course Certificate Signatories". For the "Course" option, select the `Course` that your user's `FinalGrade` is associated with. You'll need to add one or more of these to the program page in order to see certificates.

 You can navigate to a rendered certificate by visiting `http://<your_local_mm_url>/certificate/<certificate_hash>`.

#### Where should the reviewer start?
`grades/tasks.py`

#### Any background context you want to provide?
I also fixed some references to the old model name for the course certificates
